src/audio/audio_stream.c

```diff
@@ -21,10 +21,15 @@ static int audio_stream_get_buffer(struct sof_sink __sparse_cache *sink, size_t
 	struct audio_stream __sparse_cache *audio_stream =
 			attr_container_of(sink, struct audio_stream __sparse_cache,
 					  sink_api, __sparse_cache);
+	struct comp_buffer __sparse_cache *buffer_c =
+			attr_container_of(audio_stream, struct comp_buffer __sparse_cache,
+					  stream, __sparse_cache);
 
 	if (req_size > audio_stream_get_free_size(sink))
 		return -ENODATA;
 
+	buffer_stream_invalidate(buffer_c, req_size);
+
```

Comment:
@kv2019i it is already addressed, it is not unconditional
```
static inline void buffer_stream_invalidate(struct comp_buffer __sparse_cache *buffer,
					    uint32_t bytes)
{
	if (!is_coherent_shared(buffer, c))
		return;

	audio_stream_invalidate(&buffer->stream, bytes);
}

static inline void buffer_stream_writeback(struct comp_buffer __sparse_cache *buffer,
					   uint32_t bytes)
{
	if (!is_coherent_shared(buffer, c))
		return;

	audio_stream_writeback(&buffer->stream, bytes);
}
```

so those have effect in case of shared buffers only. Shared buffers are always connecting 2 cores.